authorization-server/src/modules/auth/roles-validation.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { LoggerService } from '../logger';
+import { IDidAccessTokenPayload } from './types';
+
+@Injectable()
+export class RolesValidationService {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly logger: LoggerService,
+  ) {
+    this.logger.setContext(RolesValidationService.name);
+  }
+
+  public async didAccessTokenRolesAreValid(
+    verifiedRoles: IDidAccessTokenPayload['verifiedRoles'],
+  ): Promise<boolean> {
+    const acceptedRoles =
+      this.configService.get('ACCEPTED_ROLES')?.split(',') || [];
```

Comment:
```suggestion
      this.configService.get('ACCEPTED_ROLES', []).split(',');
```
That should work as well